sound/soc/sof/ipc4-topology.c

```diff
@@ -1658,6 +1709,13 @@ static int sof_ipc4_prepare_process_module(struct snd_sof_widget *swidget,
 	if (ret < 0)
 		return ret;
 
+	if  (process->payload_with_output_fmt) {
+		ret = sof_update_hw_params(sdev, pipeline_params,
+					   &available_fmt->out_audio_fmt[i]);
```

Comment:
@libinyang how was this tested? AFAICT i is unitialized at this point. So what is available_fmt->out_audio_fmt[i]?

Reply:
@libinyang &available_fmt->out_audio_fmt[i] should be `&process->output_format`, right? IIRC, the origin PR updated pipeline_params in sof_ipc4_init_audio_fmt and it can cover all copier widgets. But now you update pipeline_params here. It means @RanderWang need to update pipeline_params in sof_ipc4_prepare_copier_module since he need to convert the format in a buffer type copier.